src/lib/Nbe.ml

```diff
@@ -793,67 +814,81 @@ struct
       quote_cof phi
     | D.TpPrf _, _ ->
       ret S.Prf
-    | _ -> 
-      Format.eprintf "bad: %a@." D.pp_con con;
-      throw @@ NbeFailed "ill-typed quotation problem"
 
-  and quote_tp_code univ =
-    function
-    | D.Nat -> 
-      ret S.Nat
-    | D.Pi (base, fam) ->
-      let+ tbase = quote_con univ base 
-      and+ tfam = 
+    | _, D.CodeNat ->
+      ret S.CodeNat
+
+    | univ, D.CodePi (base, fam) ->
+      let+ tbase = quote_con univ base
+      and+ tfam =
         let* tpbase = lift_cmp @@ do_el base in
-        binder 1 @@
-        let* var = top_var tpbase in
-        quote_con univ @<< 
-        lift_cmp @@ inst_tm_clo fam [var] 
-      in 
-      S.Pi (tbase, tfam)
-    | D.Sg (base, fam) ->
-      let+ tbase = quote_con D.Univ base 
-      and+ tfam = 
+        let+ bdy =
+          bind_var S.CofAbort tpbase @@ fun var ->
+          quote_con univ @<<
+          lift_cmp @@ do_ap fam var
+        in
+        S.Lam bdy
+      in
+      S.CodePi (tbase, tfam)
+    | univ, D.CodeSg (base, fam) ->
+      let+ tbase = quote_con univ base
+      and+ tfam =
         let* tpbase = lift_cmp @@ do_el base in
-        binder 1 @@
-        let* var = top_var tpbase in
-        quote_con univ @<< 
-        lift_cmp @@ inst_tm_clo fam [var] 
-      in 
-      S.Sg (tbase, tfam)
-    | D.Id (tp, left, right) ->
-      let* eltp = lift_cmp @@ do_el tp in
-      let+ ttp = quote_con univ tp 
-      and+ tleft = quote_con eltp left 
-      and+ tright = quote_con eltp right in 
-      S.Id (ttp, tleft, tright)
+        let+ bdy =
+          bind_var S.CofAbort tpbase @@ fun var ->
+          quote_con univ @<<
+          lift_cmp @@ do_ap fam var
+        in
+        S.Lam bdy
+      in
+      S.CodeSg (tbase, tfam)
+
+  (*
+   *  path : (fam : I -> U) -> ((i : I) -> (p : [i=0\/i=1]) -> fam i) -> U
+   * *)
+    | univ, D.CodePath (fam, bdry) -> (* check *)
+      let* tfam = quote_con univ fam in
```

Comment:
This line is incorrect, we will need to quote this thing at not `univ` but at `(i : dim) -> univ`. This type can be written down using the QQ/TB stuff (see the next line, `bdry_tp`, for an example of how to do this).